pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -442,7 +417,16 @@ def _process_single_shot(samples):
 
         self._apply_diagonalizing_gates(mps, adjoint=True)
 
-        return _process_single_shot(samples)
+        # if there is a shot vector, use the shots.bins generator to
+        # split samples w.r.t. the shots
+        processed_samples = []
+        for lower, upper in shots.bins():
+            shot = _process_single_shot(samples[..., lower:upper, :])
+            processed_samples.append(shot)
```

Comment:
is it a false positive complain?

Reply:
yeah, there are more than 100 unit tests calling this logic.

Reply:
yeah, there are more than 100+ tests triggering this part.